dsindex/indexer.go

```diff
@@ -141,56 +143,55 @@ func (x *indexer) ForEach(ctx context.Context, index string, fn func(idx, key st
 	return err
 }
 
-func (x *indexer) HasKey(ctx context.Context, index, key string) (bool, error) {
-	if index == "" {
-		return false, ErrEmptyIndex
-	}
+func (x *indexer) HasValue(ctx context.Context, key, value string) (bool, error) {
 	if key == "" {
 		return false, ErrEmptyKey
 	}
-	dskey := ds.NewKey(path.Join(x.indexPath, encode(index), encode(key)))
-	return x.dstore.Has(dskey)
+	if value == "" {
+		return false, ErrEmptyValue
+	}
+	return x.dstore.Has(ds.NewKey(encode(key)).ChildString(encode(value)))
 }
 
-func (x *indexer) HasAny(ctx context.Context, index string) (bool, error) {
+func (x *indexer) HasAny(ctx context.Context, key string) (bool, error) {
 	var any bool
-	err := x.ForEach(ctx, index, func(idx, key string) bool {
+	err := x.ForEach(ctx, key, func(key, value string) bool {
 		any = true
 		return false
 	})
 	return any, err
 }
 
-func (x *indexer) Search(ctx context.Context, index string) ([]string, error) {
-	if index == "" {
-		return nil, ErrEmptyIndex
+func (x *indexer) Search(ctx context.Context, key string) ([]string, error) {
+	if key == "" {
+		return nil, ErrEmptyKey
 	}
-	ents, err := x.queryPrefix(ctx, path.Join(x.indexPath, encode(index)))
+	ents, err := x.queryPrefix(ctx, encode(key))
 	if err != nil {
 		return nil, err
 	}
 	if len(ents) == 0 {
 		return nil, nil
 	}
 
-	keys := make([]string, len(ents))
+	values := make([]string, len(ents))
 	for i := range ents {
-		keys[i], err = decode(path.Base(ents[i].Key))
+		values[i], err = decode(path.Base(ents[i].Key))
 		if err != nil {
-			return nil, fmt.Errorf("cannot decode key: %v", err)
+			return nil, fmt.Errorf("cannot decode value: %v", err)
 		}
 	}
-	return keys, nil
+	return values, nil
 }
 
-// SyncIndex synchronizes the indexes in the target Indexer to match those of
-// the ref Indexer. The indexPath prefix is not synchronized, only the
-// index/key portion of the indexes.
+// SyncIndex synchronizes the keys in the target Indexer to match those of the
+// ref Indexer. The name portion of the stored data is not synchronized, only
```

Comment:
> The name portion

What are you trying to get at here?

Reply:
Done.

Trying to say that this function does not change the root portion of any key (the name passed into New).  Reworded to say basically that.